main.tf

```diff
@@ -20,89 +12,40 @@ data "aws_vpc" "existing" {
   multi_az = var.availability_zones > 1
   vpc_id   = var.existing_vpc_id != null ? var.existing_vpc_id : try(module.network[0].vpc_id, null)
   vpc_cidr = try(data.aws_vpc.existing[0].cidr_block, var.network_address_space)
 }
 
 module "network" {
-  source  = "terraform-aws-modules/vpc/aws"
-  version = "~> 6.0"
-  count   = var.create_network && var.existing_vpc_id == null ? 1 : 0
-
-  name = var.name
-  cidr = var.network_address_space
-
-  azs              = local.azs
-  private_subnets  = [for k, v in local.azs : cidrsubnet(var.network_address_space, 4, k)]
-  public_subnets   = [for k, v in local.azs : cidrsubnet(var.network_address_space, 8, k + 48)]
-  database_subnets = [for k, v in local.azs : cidrsubnet(var.network_address_space, 8, k + 52)]
-
-  create_database_subnet_group = false
-
-  enable_nat_gateway = true
-  single_nat_gateway = true
-
-  public_subnet_tags = {
-    "kubernetes.io/role/elb" = 1
-  }
-  private_subnet_tags = {
-    "kubernetes.io/role/internal-elb" = 1
-  }
-
-  tags = var.tags
-}
-
-module "endpoints" {
-  source  = "terraform-aws-modules/vpc/aws//modules/vpc-endpoints"
-  version = "~> 6.0"
-  count   = var.create_network && var.existing_vpc_id == null && length(var.network_private_endpoints) > 0 ? 1 : 0
-
-  vpc_id                     = local.vpc_id
-  create_security_group      = true
-  security_group_name        = "${var.name}-endpoints"
-  security_group_description = "VPC endpoint default security group"
-  security_group_rules = {
-    ingress_https = {
-      description = "HTTPS from VPC"
-      cidr_blocks = [local.vpc_cidr]
-    }
-  }
-
-  endpoints = { for endpoint_service in var.network_private_endpoints :
-    endpoint_service => {
-      service             = endpoint_service
-      subnet_ids          = module.network[0].private_subnets
-      private_dns_enabled = endpoint_service != "s3" || var.network_s3_private_dns_enabled
-      dns_options = {
-        private_dns_only_for_inbound_resolver_endpoint = false
-      }
-    }
-  }
-
-  tags = var.tags
-}
-
-resource "aws_route53_record" "s3_endpoint_cname" {
-  count = var.create_network && var.existing_vpc_id == null && contains(var.network_private_endpoints, "s3") && !var.network_s3_private_dns_enabled ? 1 : 0
-
-  zone_id = local.private_zone_id
-  name    = var.fips_enabled ? "s3-fips.${data.aws_region.current.region}.amazonaws.com" : "s3.${data.aws_region.current.region}.amazonaws.com"
-  type    = "CNAME"
-  records = [module.endpoints[0].endpoints["s3"].dns_entry[0].dns_name]
-  ttl     = 300
-}
-
-module "flow_log" {
-  source = "terraform-aws-modules/vpc/aws//modules/flow-log"
-  count  = var.network_enable_vpc_flow_logs ? 1 : 0
-
-  name   = var.name
-  vpc_id = local.vpc_id
-
-  cloudwatch_log_group_use_name_prefix   = false
-  cloudwatch_log_group_retention_in_days = var.network_cloudwatch_log_group_retention_in_days
+  source = "./modules/network"
+  count  = var.create_network && var.existing_vpc_id == null ? 1 : 0
```

Comment:
This won't allow to create endpoints when using existing vpc. This is expected or we should have possibility to create vpc endpoints when using existing vpc?